datadog_checks_dev/datadog_checks/dev/docker.py

```diff
@@ -50,21 +50,35 @@ def compose_file_active(compose_file):
 
 @contextmanager
 def shared_logs(example_log_configs, mount_whitelist=None):
+    def normalize(s):
+        """Make sure 's' can be used as part of a file name and an environment variable name."""
+        return s.lower().replace(' ', '_')
+
     log_source = example_log_configs[0].get('source', 'check')
 
-    if mount_whitelist is None:
-        # Default to all
-        mount_whitelist = range(1, len(example_log_configs) + 1)
+    if mount_whitelist is not None:
+        mount_whitelist = [normalize(alias) for alias in mount_whitelist]
 
     env_vars = {}
     docker_volumes = get_state('docker_volumes', [])
 
     with ExitStack() as stack:
-        for i, example_log_config in enumerate(example_log_configs, 1):
-            if i not in mount_whitelist:
+        for example_log_config in example_log_configs:
+            try:
+                alias = example_log_config["alias"]
+            except KeyError:
```

Comment:
Instead of raising an error if there is no alias, could we default to using `DD_LOG_1` so that we could still run the environment with logs?